### DISCUSSION TITLE
I have an svg file and want to convert into png or eps using imagick function

### DISCUSSION THREAD

**Discussion by sananisoft** [ImageMagick/ImageMagick#6266] [Help]
### ImageMagick version

6.9.11-60 Q16 x86_64 2021-01-25

### Operating system, version and so on

Ubuntu 22.04.2 LTS

### Description

I have an svg file and want to convert into png or eps using imagick function
It converts into png successfully but converted png is  not according to svg. It only shows the background  of the image, and the actual image is missing.
here the raw png
<?xml version="1.0" ?><svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0" y="0" width="752" height="752" viewBox="0 0 512 512" enable-background="new 0 0 500 500" xml:space="preserve" data-v-2bbc45f4=""><circle r="256" cx="256" cy="256" fill="#0086F0" shape="circle" data-v-2bbc45f4=""></circle><g data-v-2bbc45f4="">
    
    <svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
         viewBox="0 0 752 752" style="enable-background:new 0 0 752 752;" xml:space="preserve" stroke="#FA0606" stroke-width="12pt">
    <path transform-origin="center" transform="rotate(-270)" d="M578.4,513.3c0,67.1-54.6,121.7-121.7,121.7c-35.7,0-67.9-15.5-90.2-40H222.7c-27.1,0-49.1-22-49.1-49.1V166.1
        c0-27.1,22-49.1,49.1-49.1h188.9c0.8,0,1.6,0.3,2.1,0.9L517.8,222c0.6,0.6,0.9,1.3,0.9,2.1v184.4
        C554.4,429.8,578.4,468.8,578.4,513.3z M414.5,178.1c0,23.8,19.3,43.1,43.1,43.1h50.8l-93.9-93.9V178.1z M361.5,589
        c-16.5-20.8-26.4-47.1-26.4-75.7c0-67.1,54.6-121.7,121.7-121.7c20.2,0,39.2,4.9,56,13.7V227.2h-55.1c-27.1,0-49.1-22-49.1-49.1V123
        H222.7c-23.8,0-43.1,19.3-43.1,43.1v379.8c0,23.8,19.3,43.1,43.1,43.1H361.5z M572.4,513.3c0-63.8-51.9-115.7-115.7-115.7
        S341,449.5,341,513.3S392.9,629,456.7,629S572.4,577.1,572.4,513.3z M513.1,510.3h-53.4v-53.4c0-1.7-1.3-3-3-3s-3,1.3-3,3v53.4
        h-53.4c-1.7,0-3,1.3-3,3s1.3,3,3,3h53.4v53.4c0,1.7,1.3,3,3,3s3-1.3,3-3v-53.4h53.4c1.7,0,3-1.3,3-3S514.8,510.3,513.1,510.3z"/>
    </svg>
    </g> </svg>
Actual Image look like this
![image](https://user-images.githubusercontent.com/121680928/233335406-18196dea-ffc6-4454-afd9-e62404490be2.png)
https://user-images.githubusercontent.com/121680928/233335406-18196dea-ffc6-4454-afd9-e62404490be2.png

And Library convert After
![image](https://user-images.githubusercontent.com/121680928/233336183-fa12a65c-f9a4-41e9-9726-fef9ae227c31.png)
https://user-images.githubusercontent.com/121680928/233336183-fa12a65c-f9a4-41e9-9726-fef9ae227c31.png

And here my PHP code
im using php version 8.1
 $im = new \Imagick();
                        $im->readImageBlob($dynamicSvg);
                        /*png settings*/
                        $im->setImageFormat("PNG24");
                        $im->setOption('png:bit-depth', '16');
                        $im->writeImage($dynamicPath . '/' . $key . 'one.png');
                        $im->clear();
                        $im->destroy();

**Comment by snibgo** [comment]
Your SVG is bad: there should be nothing after `</svg>`.

If you fix that and then open the SVG in Inkscape, you will see the drawing is entirely outside the page. This can be fixed by editing the SVG: change "`rotate(-270)`" to "`translate (700,0) rotate(-270)`".

**Reply by sananisoft** [reply]
<?xml version="1.0" encoding="UTF-8"?><svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0" y="0" width="752" height="752" viewBox="0 0 512 512" enable-background="new 0 0 500 500" xml:space="preserve" data-v-2bbc45f4="">
        <circle r="256" cx="256" cy="256" fill="#0086f0" shape="circle" data-v-2bbc45f4=""/>
        <g data-v-2bbc45f4="">
          <svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px" viewBox="0 0 752 752" style="enable-background:new 0 0 752 752;" xml:space="preserve" stroke="#e4a230" stroke-width="12pt">
            <path transform-origin="center" transform="rotate(-270)" d="M579.3,466.7c-1.6,0-3.2,0.1-4.8,0.2c-1.7-11.9-7.5-22.9-16.3-31c-9.8-9.1-22.5-14.1-35.9-14.1c-11,0-21.9,3.5-30.8,9.9
          L358.2,200.9c-7.7-13.3-21.3-21.2-36.6-21.2s-29,7.9-36.6,21.2L125.7,476.8c-7.6,13.2-7.6,29.1,0,42.3c7.7,13.2,21.3,21.2,36.6,21.2
          h223.2c8.3,19.2,27.6,32,48.5,32h145.2c29.1,0,52.7-23.7,52.7-52.7C632,490.4,608.3,466.7,579.3,466.7z M313.8,274.5
          c0-4.3,3.5-7.8,7.8-7.8c4.3,0,7.8,3.5,7.8,7.8v155.2c0,4.3-3.5,7.8-7.8,7.8c-4.3,0-7.8-3.5-7.8-7.8V274.5z M321.7,463.3
          c-4.3,0-7.9-3.5-7.9-7.8c0-4.3,3.5-7.8,7.9-7.8c4.3,0,7.8,3.5,7.8,7.8C329.6,459.8,326,463.3,321.7,463.3z M579.3,556.5H434.1
          c-20.4,0-37-16.6-37-37.1c0-15.3,9.6-29.2,23.8-34.6c2.4-0.9,4.2-3,4.8-5.5c4-16.7,18.8-28.3,36-28.3c6.8,0,13.4,1.9,19.2,5.4
          c3.6,2.2,8.3,1.2,10.6-2.4c6.9-10.4,18.4-16.5,30.9-16.5c20.4,0,37.1,16.6,37.1,37c0,0.4,0,0.9,0,1.3c-0.1,2.5,1,4.9,3,6.5
          c2,1.6,4.6,2.1,7,1.4c3.2-0.9,6.6-1.3,10-1.3c20.4,0,37,16.6,37,37C616.3,539.9,599.7,556.5,579.3,556.5z"/>
          </svg>
        </g>
      </svg>
      i have this svg this also doing same thing

**Comment by snibgo** [comment]
Where did the SVG come from? Like your previous example, it has bad syntax.
